src/driver/express/ExpressDriver.ts

```diff
@@ -117,15 +118,19 @@ export class ExpressDriver extends BaseDriver implements Driver {
                 if (isPromiseLike(checkResult)) {
                     checkResult.then(result => {
                         if (!result) {
-                            return this.handleError(new AccessDeniedError(action), actionMetadata, action);
+                            return this.handleError((actionMetadata.authorizedRoles.length === 0
+                                    ? new AuthorizationRequiredError(action) : new AccessDeniedError(action)),
+                                actionMetadata, action);
 
                         } else {
                             next();
                         }
                     });
                 } else {
                     if (!checkResult) {
-                        this.handleError(new AccessDeniedError(action), actionMetadata, action);
+                        this.handleError((actionMetadata.authorizedRoles.length === 0
```

Comment:
> so if the libs want's to be fast

I see you need time to heal from the traumatic experience of that conversation 😄